src/api/libcellml/when.h

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright libCellML Contributors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#pragma once
+
+#include <string>
+
+#include "libcellml/exportdefinitions.h"
+#include "libcellml/entity.h"
+#include "libcellml/orderedentity.h"
+#include "libcellml/types.h"
+
+namespace libcellml {
+
+/**
+ * @brief The When class.
+ * Class for determining when a reset will occur.
+ */
+class LIBCELLML_EXPORT When: public Entity, public OrderedEntity
+{
+public:
+    When(); /**< Constructor */
+    ~When(); /**< Destructor */
+    When(const When &rhs); /**< Copy constructor */
+    When(When &&rhs); /**< Move constructor */
+    When& operator=(When n); /**< Assignment operator */
+
+    /**
+     * @brief Set the condition term for this when.
+     *
+     * Set the @c std::string MathML string as the condition term for this when.
+     *
+     * @param condition The @c std::string MathML string of the condition to set.
+     */
+    void setCondition(const std::string& condition);
```

Comment:
Minor: there should be a space before `&` and none after, i.e.

void setCondition(const std::string &condition);

The same is true in various parts of the code in fact (e.g. `setValue()` below).